sycl/source/detail/fusion/fusion_wrapper_impl.cpp

```diff
@@ -20,22 +22,23 @@ std::shared_ptr<detail::queue_impl> fusion_wrapper_impl::get_queue() const {
   return MQueue;
 }
 
-bool fusion_wrapper_impl::is_in_fusion_mode() const { return false; }
+bool fusion_wrapper_impl::is_in_fusion_mode() const {
+  auto QUniqueID = std::hash<QueueImplPtr>()(MQueue);
+  return detail::Scheduler::getInstance().isInFusionMode(QUniqueID);
```

Comment:
Alternatively:
```suggestion
  return MQueue->is_in_fusion_mode();
```
Not a huge difference, but a little less code to maintain. 😄